### DISCUSSION TITLE
ConTeXt: cannot pass options to \inputmarkdown

### DISCUSSION THREAD

**Discussion by ibotty** [Witiko/markdown#389] [Q&A] [answered]
I am using markdown from the `texlive-markdown-svn66257-69.fc39.noarch` package.  I don't know how to get the markdown version.

The document
```
\usemodule[t][markdown]
\starttext
\inputmarkdown[smart_ellipses]{input1.md}
\stoptext
```
fails with
```
> context test.mkiv
[...]
tex error       > tex error on line 3 in file /var/home/tob/src/zebra/pdf-generator/examples/output.mkiv: ! Use of \setupmarkdown doesn't match its definition


\doinputmarkdown [#1]#2->\begingroup \iffirstargument \setupmarkdown {
                                                                      #1}\fi \markdownInput {#2}\endgroup
l.3 \inputmarkdown[smart_ellipses]{input1.md}
[...]
```

It Removing the argument (`[smart_ellipses]`) it compiles and renders markdown.

Using `\setupmarkdown` let's me pass options.  I'd like to use different options when including markdown though.

**Comment by Witiko** [comment, marked as answer]
Hi, thanks for using the Markdown package.

> I don't know how to get the markdown version.

You can find out as follows:

``` tex
\usemodule[t][markdown]
\starttext
Markdown version: \markdownVersion
\stoptext
```

> The document [...] fails with [...]
> Removing the argument (`[smart_ellipses]`) it compiles and renders markdown.

This is a bug, thanks for reporting it. It has been fixed in #390 and will be released in Markdown 3.3.0 later this month. The bugfix will be part of the TeX Live 2024 distribution.

> Using \setupmarkdown let's me pass options. I'd like to use different options when including markdown though.

You can do this manually as follows:

``` tex
\usemodule[t][markdown]
\starttext
\begingroup
\setupmarkdown[smart_ellipses]
\inputmarkdown{input1.md}
\endgroup
\stoptext
```

Here, the option `smart_ellipses` will only be enabled within `\begingroup ... \endgroup`.

**Reply by ibotty** [reply]
Thank you for the prompt fix and the workaround!  Very appreciated.
